partiql-parser/src/main/antlr/PartiQL.g4

```diff
@@ -334,6 +334,20 @@ windowSortSpecList
 havingClause
     : HAVING arg=exprSelect;
 
+excludeClause
+    : EXCLUDE excludeExpr (COMMA excludeExpr)*;
+
+excludeExpr
+    : symbolPrimitive excludeExprSteps+;  // Require 1 more `excludeExprSteps`. Disallow `EXCLUDE a`
+
+excludeExprSteps
+    : PERIOD symbolPrimitive                            # ExcludeExprTupleAttr
+    | BRACKET_LEFT attr=LITERAL_STRING BRACKET_RIGHT    # ExcludeExprCollectionAttr
+    | BRACKET_LEFT index=LITERAL_INTEGER BRACKET_RIGHT  # ExcludeExprCollectionIndex
+    | BRACKET_LEFT ASTERISK BRACKET_RIGHT               # ExcludeExprCollectionWildcard
+    | PERIOD ASTERISK                                   # ExcludeExprTupleWildcard
+    ;
```

Comment:
Can you give me a breakdown of what these semantically mean for your PR?

If I'm guessing:
- `EXCLUDE t.*` (when `t` is a struct) will make `t` an empty struct?
- `EXCLUDE t[*]` (when `t` is a list) will make `t` an empty list?
- `EXCLUDE t[2]` (when `t` is a list) will remove the item at index 2
- `EXCLUDE t[*].a` (when `t` is a list of structs) will remove attribute a from all items in `t`.

Follow up questions:
- Is `t.*.*[1]` allowed?

Reply:
Yes to all.